qutip/solver/parallel.py

```diff
@@ -369,13 +383,11 @@ def loky_pmap(task, values, task_args=None, task_kwargs=None,
         [job.cancel() for job in jobs]
         raise e
 
-    except TimeoutError:
-        [job.cancel() for job in jobs]
-
     finally:
-        executor.shutdown()
+        os.environ['QUTIP_IN_PARALLEL'] = 'FALSE'
+        executor.shutdown(kill_workers=True)
```

Comment:
What does `kill_workers=True` do?

Reply:
It kills all worker processes immediately. (That might sound worrisome since killing threads forcibly is a _bad thing_ generally, but these are actual processes, so it is okay afaik.)

In `parallel_map`, if there is an error (and `fail_fast` enabled) or if we hit the timeout, we still cannot return until all currently running tasks finish, since the standard ProcessPoolExecutor does not have such an option. The loky version offers that `kill_workers` flag, making it possible to return immediately in these cases, which seemed useful to me.

Reply:
I went and read the loky docs and this use of it seems okay. And it is really nice to not have to wait for long worker tasks to complete.